github4s/js/src/test/scala/github4s/utils/integration/GHStatusesSpec.scala

```diff
@@ -0,0 +1,103 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package github4s.utils.integration
+
+import github4s.Decoders._
+import github4s.Github
+import github4s.Github._
+import github4s.free.domain.{CombinedStatus, Status}
+import github4s.js.Implicits._
+import github4s.utils.TestUtils
+import org.scalatest.{AsyncFlatSpec, Matchers}
+
+class GHStatusesSpec extends AsyncFlatSpec with Matchers with TestUtils {
+
+  override implicit val executionContext = scala.concurrent.ExecutionContext.Implicits.global
+
+  "Statuses >> Get" should "return a combined status" in {
+    val response = Github(accessToken).statuses
+      .getCombinedStatus(validRepoOwner, validRepoName, validRefSingle)
+      .execFuture(headerUserAgent)
+
+    testFutureIsRight[CombinedStatus](response, { r =>
+      r.result.repository.full_name shouldBe s"$validRepoOwner/$validRepoName"
+      r.statusCode shouldBe okStatusCode
+    })
+  }
+
+  it should "return an error when an invalid ref is passed" in {
+    val response = Github(accessToken).statuses
+      .getCombinedStatus(validRepoOwner, validRepoName, invalidRef)
+      .execFuture(headerUserAgent)
+    response should be('left)
+  }
+
+  "Statuses >> List" should "return a non empty list when a valid ref is provided" in {
+    val response = Github(accessToken).statuses
+      .listStatuses(validRepoOwner, validRepoName, validRefSingle)
+      .execFuture(headerUserAgent)
+
+    testFutureIsRight[List[Status]](response, { r =>
+      r.result.nonEmpty shouldBe true
+      r.statusCode shouldBe okStatusCode
+    })
+  }
+
+  it should "return an empty list when an invalid ref is provided" in {
+    val response = Github(accessToken).statuses
+      .listStatuses(validRepoOwner, validRepoName, invalidRef)
+      .execFuture(headerUserAgent)
+
+    testFutureIsRight[List[Status]](response, { r =>
+      r.result.isEmpty shouldBe true
+      r.statusCode shouldBe okStatusCode
+    })
+  }
+
+  "Status >> Create" should "create a status" in {
+    val response = Github(accessToken).statuses
+      .createStatus(
+        "BenFradet",
+        "sparkml-als",
+        "ae6eb4ae126f29d96249a2393c633c8d91a4adf4",
+        validStatusState,
+        None,
+        None,
+        None)
+      .execFuture(headerUserAgent)
+
+    testFutureIsRight[Status](response, { r =>
+      r.result.state shouldBe validStatusState
+      r.statusCode shouldBe createdStatusCode
+    })
+  }
+
+  it should "return an error when an invalid sha is passed" in {
+    val response = Github(accessToken).statuses
+      .createStatus(
+        "BenFradet",
+        "sparkml-als",
+        invalidCommitSha,
+        validStatusState,
+        None,
+        None,
+        None)
+      .execFuture(headerUserAgent)
+
+    testFutureIsLeft(response)
+  }
```

Comment:
We've tested it locally, we know it fails with fork but it should work once it's merged